File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Daria/Laborator_10/decorator/ExtraCheese.java
package Laborator.Daria.Laborator_10.decorator;

public class ExtraCheese extends PizzaDecorator {   
    public ExtraCheese(Pizza pizza) {
        super(pizza);
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", Extra Cheese";
    }

    @Override
    public double getCost() {
        return super.getCost() + 5.0;
    }
}

File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Paul/design-patterns/decorator/src/main/kotlin/pizza/ingredients/ExtraCheese.kt
package org.paul.pizza.ingredients

import org.paul.pizza.Pizza
import org.paul.pizza.PizzaDecorator

class ExtraCheese(pizza: Pizza) : PizzaDecorator(pizza) {
    override fun getCost(): Double {
        return super.getCost() + COST
    }

    override fun getDescription(): String {
        return super.getDescription() + ", Extra Cheese"
    }

    companion object {
        private const val COST = 5.0
    }
}
